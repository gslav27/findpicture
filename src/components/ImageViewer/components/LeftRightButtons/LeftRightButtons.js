import React from 'react';
import classNames from 'classnames';

import { withStyles } from '@material-ui/core/styles';

import IconButton from '@material-ui/core/IconButton';

import ChevronLeft from '@material-ui/icons/ChevronLeft';
import ChevronRight from '@material-ui/icons/ChevronRight';

import leftRightButtonsStyles from './LeftRightButtons.styles';


const styles = leftRightButtonsStyles;


const LeftRightButtons = (props) => {
  const {
    mobileWithTouch,
    classes,
  } = props;

  return (
    <div className={classes.viewerAddsNavigation}>
      <IconButton
        onClick={() => props.onChange('left')}
        classes={{
          root: classNames(classes.viewerIconButtonLR, { [classes.buttonsHover]: !mobileWithTouch }),
          label: classes.viewerIconButtonLabel,
        }}
        disableRipple
        title='left'
      >
        <ChevronLeft className={classes.viewerIconStyleLR} />
      </IconButton>
      <IconButton
        onClick={() => props.onChange('right')}
        className={classes.viewerIconButtonLabelRight}
        classes={{
          root: classNames(classes.viewerIconButtonLR, { [classes.buttonsHover]: !mobileWithTouch }),
          label: classes.viewerIconButtonLabel,
        }}
        disableRipple
        title='right'
      >
        <ChevronRight className={classes.viewerIconStyleLR} />
      </IconButton>
    </div>
  );
};

export default withStyles(styles)(LeftRightButtons);
