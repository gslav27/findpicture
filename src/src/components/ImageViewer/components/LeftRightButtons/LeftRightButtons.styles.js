import imageViewerStyles from '../../ImageViewer.styles';

const style = {
  viewerAddsNavigation: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    height: '100%',
  },
  viewerIconButtonLR: {
    display: 'block',
    boxSizing: 'border-box',
    width: '45vw',
    height: '40vh',
    padding: 3,
    cursor: 'pointer',
    '&:hover': { backgroundColor: 'rgba(0,0,0,0)' },
    zIndex: 1502,
  },
  viewerIconButtonLabel: { width: '10vw' },
  viewerIconButtonLabelRight: { '& span': { float: 'right' } },
  viewerIconStyleLR: {
    width: '1.3em',
    height: '1.3em',
    maxHeight: 50,
    color: '#ccc',
  },
  ...imageViewerStyles(),
};

export default style;
