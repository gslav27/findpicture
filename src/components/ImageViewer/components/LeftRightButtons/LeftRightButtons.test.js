import React from 'react';
import renderer from 'react-test-renderer';
import LeftRightButtons from './LeftRightButtons';

describe('<LeftRightButtons />', () => {
  describe('when "mobileWithTouch" props is true', () => {
    test('should render component without "buttonsHover" class for all buttons', () => {
      const tree = renderer.create(<LeftRightButtons mobileWithTouch />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('when "mobileWithTouch" props is false', () => {
    test('should render component with "buttonsHover" class for all buttons', () => {
      const tree = renderer.create(<LeftRightButtons mobileWithTouch={false} />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
