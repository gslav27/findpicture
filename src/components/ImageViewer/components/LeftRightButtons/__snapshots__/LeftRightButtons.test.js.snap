// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LeftRightButtons /> when "mobileWithTouch" props is false should render component with "buttonsHover" class for all buttons 1`] = `
<div
  className="LeftRightButtons-viewerAddsNavigation-1"
>
  <button
    className="MuiButtonBase-root-27 MuiIconButton-root-21 LeftRightButtons-viewerIconButtonLR-2 LeftRightButtons-buttonsHover-7"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    title="left"
    type="button"
  >
    <span
      className="MuiIconButton-label-26 LeftRightButtons-viewerIconButtonLabel-3"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-30 LeftRightButtons-viewerIconStyleLR-5"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
          />
        </g>
      </svg>
    </span>
  </button>
  <button
    className="MuiButtonBase-root-27 MuiIconButton-root-21 LeftRightButtons-viewerIconButtonLR-2 LeftRightButtons-buttonsHover-7 LeftRightButtons-viewerIconButtonLabelRight-4"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    title="right"
    type="button"
  >
    <span
      className="MuiIconButton-label-26 LeftRightButtons-viewerIconButtonLabel-3"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-30 LeftRightButtons-viewerIconStyleLR-5"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </g>
      </svg>
    </span>
  </button>
</div>
`;

exports[`<LeftRightButtons /> when "mobileWithTouch" props is true should render component without "buttonsHover" class for all buttons 1`] = `
<div
  className="LeftRightButtons-viewerAddsNavigation-1"
>
  <button
    className="MuiButtonBase-root-27 MuiIconButton-root-21 LeftRightButtons-viewerIconButtonLR-2"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    title="left"
    type="button"
  >
    <span
      className="MuiIconButton-label-26 LeftRightButtons-viewerIconButtonLabel-3"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-30 LeftRightButtons-viewerIconStyleLR-5"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
          />
        </g>
      </svg>
    </span>
  </button>
  <button
    className="MuiButtonBase-root-27 MuiIconButton-root-21 LeftRightButtons-viewerIconButtonLR-2 LeftRightButtons-viewerIconButtonLabelRight-4"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    title="right"
    type="button"
  >
    <span
      className="MuiIconButton-label-26 LeftRightButtons-viewerIconButtonLabel-3"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-30 LeftRightButtons-viewerIconStyleLR-5"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </g>
      </svg>
    </span>
  </button>
</div>
`;
